docs/src/inverse_problems/petab_ode_param_fitting.md

```diff
@@ -0,0 +1,494 @@
+2. Gradient in the `petab_problem.compute_gradient!` field. It takes two arguments (`g` and `p`) with the updated gradient values being written to `g`.
+3. Hessian in the `petab_problem.compute_hessian` field. It takes two arguments (`H` and `p`) with the updated hessian values being written to `H`.
+
+
+## [Multi-start optimisation](@id petab_multistart_optimisation)
+To avoid the optimisation process returning a local minimum, it is often advised to run it multiple times, using different initial guesses. PEtab.jl supports this through the `calibrate_model_multistart` function. This is identical to the `calibrate_model` function, but takes two additional arguments:
+
+1. `n_multistart`: The number of runs to perform.
+2. `dir_save`: A location to which the output is saved. If `dir_save=nothing`, no output is saved.
+
+And one additional optional argument:
+
+3. `sampling_method`: Selects the sampling method with which to select the initial guesses (`QuasiMonteCarlo.LatinHypercubeSample()` used by default). 
+ 
+Because `calibrate_model_multistart` handles initial guess sampling, unlike for `calibrate_model`, no initial guess has to be provided. 
+
+Here, we fit parameters through 10 independent optimisation runs, using QuasiMonteCarlo's `SobolSample` method, and save the result to the OptimizationRuns folder:
+```@example petab1
+using Optim
+using QuasiMonteCarlo
+res_ms = calibrate_model_multistart(petab_problem, IPNewton(), 10, "OptimizationRuns"; sampling_method=QuasiMonteCarlo.SobolSample())
+```
+The best result across all runs can still be retrieved using `get_ps(res_ms, petab_problem)`, with the results of the individual runs being stored in the `res_ms.runs` field. 
+
+To load the result in a later session, we can call:
+```@example petab1
+res_ms = PEtabMultistartOptimisationResult("OptimizationRuns")
+```
+If the OptimizationRuns folder contains the output from several runs, we can designate which to load using the `which_run` argument. Here we loads the second run to be saved in that folder:
+```@example petab1
+res_ms = PEtabMultistartOptimisationResult("OptimizationRuns"; which_run="2")
+```
+
+
+## [Events](@id petab_events)
+So far, we have assumed that all experiments, after initiation, run without interference. Experiments where conditions change, or where species are added/removed during the time course, can be represented through events (related to [callbacks](@ref advanced_simulations_callbacks)). In PEtab, an event is represented through the `PEtabEvent` structure. It takes three arguments:
+1. The condition for triggering the event. This can either indicate a point in time, or a boolean condition.
+2. A rule for updating the event's target
+3. The event's target (either a species or parameter).
+
+Here we create an event which adds `0.5` units of `S` to the system at time `5.0`:
+```@example petab1
+@unpack S = rn
+event1 = PEtabEvent(5.0, S + 0.5, S)
+```
```

Comment:
This should instead be put into the Catalyst model and forwarded appropriately. Does that work?

Reply:
I suggest putting the events into the Catalyst model and then just showing it forwards ok in this section. You can then refer to the PEtab docs for adding events in their interface.

Reply:
Currently it the optimal way is to use PEtab's interface. I have forgotten if it had to do with the performance or limitations of the SciML events (@sebapersson can probably expand). If I remember right, the idea is to, eventually, add a routine that detects events in Catalyst models and translates them into the PEtab format.
(essentially giving this effect)

Reply:
What are the limitations / performance issues? 

As I said on Slack, I think there needs to be an explicit bolded/italicized statement then that PETab will ignore events in Catalyst models, so users must then reenter them manually in this way. Otherwise this potentially creates confusion that events in models are being silently dropped.

Reply:
I have added a specific note box on this.